File: Python/007.py
from math import sqrt

def is_prime(n):
    for i in range(2, int(sqrt(n)) + 1):
        if n % i == 0:
            return False
    return True

def solve(nth):
    n, count = 2, 0
    while count < nth:
        if is_prime(n):
            count += 1
        n += 1
    return n - 1

# n = 6
n = 10001
print(solve(n))

File: C++/007.cpp
#include <iostream>
#include <cmath>
using namespace std;

bool is_prime(int n)
{
    for (int i = 2; i <= (int)sqrt(n); i++)
        if (n % i == 0)
            return false;
    return true;
}

int solve(int nth)
{
    int n = 2, count = 0;
    while (count < nth) {
        if (is_prime(n))
            count++;
        n++;
    }
    return n - 1;
}

int main()
{
    // int n = 6;
    int n = 10001;
    cout << solve(n) << endl;
}
